LogMonitor/src/LogMonitor/Main.cpp

```diff
@@ -249,6 +270,14 @@ int __cdecl wmain(int argc, WCHAR *argv[])
         logWriter.TraceError(L"Invalid configuration file.");
     }
 
+    //
+    // Set up a signal handler for common catastrophic failure signals
+    //
+    setSignalHandler(SIGSEGV, signalHandler); // Segmentation fault
+    setSignalHandler(SIGFPE, signalHandler);  // Floating point exception
+    setSignalHandler(SIGILL, signalHandler);  // Illegal instruction
+    setSignalHandler(SIGABRT, signalHandler); // Abort

```

Comment:
This looks good to me. There are more signals like SIGBUS and SIGPIPE etc that might cause catastrophic failures do you reckon we should include them? Also you can think if we can capture more info when we have such failures. At the moment we are only logging the signal type. It is possbile to log info like timestamp, stack trace when we have these signals?

Reply:
Using the TraceError function to log the catastrophic failure ensures that timestamp is captured: 

https://github.com/microsoft/windows-container-tools/blob/04885c694bb89ca78437f8dcfe19389f3211102b/LogMonitor/src/LogMonitor/LogWriter.h#L93C13-L93C53

![image](https://github.com/microsoft/windows-container-tools/assets/28774968/d2ce2e9b-b88a-4465-b10f-03c343bd14a1)